// node-alarm-dot-com/dist/_models/SensorType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SensorType = void 0;
/* eslint-disable @typescript-eslint/naming-convention */
/**
 * Number indicating the type of sensor according to the Alarm.com API
 */
var SensorType;
(function (SensorType) {
    SensorType[SensorType["Contact_Sensor"] = 1] = "Contact_Sensor";
    SensorType[SensorType["Motion_Sensor"] = 2] = "Motion_Sensor";
    SensorType[SensorType["Smoke_Detector"] = 5] = "Smoke_Detector";
    SensorType[SensorType["CO_Detector"] = 6] = "CO_Detector";
    SensorType[SensorType["Heat_Detector"] = 8] = "Heat_Detector";
    SensorType[SensorType["Fob"] = 9] = "Fob";
    SensorType[SensorType["Keypad"] = 10] = "Keypad";
    SensorType[SensorType["Water_Sensor"] = 17] = "Water_Sensor";
    SensorType[SensorType["Glass_Break"] = 19] = "Glass_Break";
    SensorType[SensorType["Panel_Camera"] = 68] = "Panel_Camera";
    SensorType[SensorType["Panel_Glass_Break"] = 83] = "Panel_Glass_Break";
})(SensorType || (exports.SensorType = SensorType = {}));
