// node-alarm-dot-com/dist/_models/States.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.REL_TYPES = exports.THERMOSTAT_STATES = exports.GARAGE_STATES = exports.LOCK_STATES = exports.LIGHT_STATES = exports.SENSOR_STATES = exports.SYSTEM_STATES = void 0;
/* eslint-disable @typescript-eslint/naming-convention */
/***
 * State of the partition as defined by Alarm.com
 * @readonly
 * @enum {number}
 */
var SYSTEM_STATES;
(function (SYSTEM_STATES) {
    SYSTEM_STATES[SYSTEM_STATES["UNKNOWN"] = 0] = "UNKNOWN";
    SYSTEM_STATES[SYSTEM_STATES["DISARMED"] = 1] = "DISARMED";
    SYSTEM_STATES[SYSTEM_STATES["ARMED_STAY"] = 2] = "ARMED_STAY";
    SYSTEM_STATES[SYSTEM_STATES["ARMED_AWAY"] = 3] = "ARMED_AWAY";
    SYSTEM_STATES[SYSTEM_STATES["ARMED_NIGHT"] = 4] = "ARMED_NIGHT";
})(SYSTEM_STATES || (exports.SYSTEM_STATES = SYSTEM_STATES = {}));
/***
 * State of the sensor as defined by Alarm.com
 * @readonly
 * @enum {number}
 */
var SENSOR_STATES;
(function (SENSOR_STATES) {
    SENSOR_STATES[SENSOR_STATES["UNKNOWN"] = 0] = "UNKNOWN";
    SENSOR_STATES[SENSOR_STATES["CLOSED"] = 1] = "CLOSED";
    SENSOR_STATES[SENSOR_STATES["OPEN"] = 2] = "OPEN";
    SENSOR_STATES[SENSOR_STATES["IDLE"] = 3] = "IDLE";
    SENSOR_STATES[SENSOR_STATES["ACTIVE"] = 4] = "ACTIVE";
    SENSOR_STATES[SENSOR_STATES["DRY"] = 5] = "DRY";
    SENSOR_STATES[SENSOR_STATES["WET"] = 6] = "WET";
})(SENSOR_STATES || (exports.SENSOR_STATES = SENSOR_STATES = {}));
/***
 * State of the light as defined by Alarm.com
 * @readonly
 * @enum {number}
 */
var LIGHT_STATES;
(function (LIGHT_STATES) {
    LIGHT_STATES[LIGHT_STATES["ON"] = 2] = "ON";
    LIGHT_STATES[LIGHT_STATES["OFF"] = 3] = "OFF";
})(LIGHT_STATES || (exports.LIGHT_STATES = LIGHT_STATES = {}));
/***
 * State of the lock as defined by Alarm.com
 * @readonly
 * @enum {number}
 */
var LOCK_STATES;
(function (LOCK_STATES) {
    LOCK_STATES[LOCK_STATES["SECURED"] = 1] = "SECURED";
    LOCK_STATES[LOCK_STATES["UNSECURED"] = 2] = "UNSECURED";
})(LOCK_STATES || (exports.LOCK_STATES = LOCK_STATES = {}));
/***
 * State of the garage as defined by Alarm.com
 * @readonly
 * @enum {number}
 */
var GARAGE_STATES;
(function (GARAGE_STATES) {
    //UNKNOWN: 0,  //ADC does not have an unknown state. ADC returns temp popup
    GARAGE_STATES[GARAGE_STATES["OPEN"] = 1] = "OPEN";
    GARAGE_STATES[GARAGE_STATES["CLOSED"] = 2] = "CLOSED"; //double check
})(GARAGE_STATES || (exports.GARAGE_STATES = GARAGE_STATES = {}));
/***
 * State of the thermostat as defined by Alarm.com
 * @readonly
 * @enum {number}
 */
var THERMOSTAT_STATES;
(function (THERMOSTAT_STATES) {
    THERMOSTAT_STATES[THERMOSTAT_STATES["OFF"] = 1] = "OFF";
    THERMOSTAT_STATES[THERMOSTAT_STATES["HEATING"] = 2] = "HEATING";
    THERMOSTAT_STATES[THERMOSTAT_STATES["COOLING"] = 3] = "COOLING";
    THERMOSTAT_STATES[THERMOSTAT_STATES["AUTO"] = 4] = "AUTO";
})(THERMOSTAT_STATES || (exports.THERMOSTAT_STATES = THERMOSTAT_STATES = {}));
/***
 * Relation types as defined by Alarm.com.
 * Relationship types tell you what object is being sent from the API.
 * @readonly
 */
var REL_TYPES;
(function (REL_TYPES) {
    REL_TYPES["CONFIGURATION"] = "systems/configuration";
    REL_TYPES["PARTITION"] = "devices/partition";
    REL_TYPES["SENSOR"] = "devices/sensor";
    REL_TYPES["LIGHT"] = "devices/light";
    REL_TYPES["LOCK"] = "devices/lock";
    REL_TYPES["GARAGE_DOOR"] = "devices/garage-door";
    REL_TYPES["CAMERA"] = "video/camera";
    REL_TYPES["THERMOSTAT"] = "devices/thermostat";
    REL_TYPES["GEO_DEVICE"] = "geolocation/geo-device";
    REL_TYPES["GEO_FENCE"] = "geolocation/fence";
    REL_TYPES["SCENE"] = "automation/scene";
})(REL_TYPES || (exports.REL_TYPES = REL_TYPES = {}));
