// node-alarm-dot-com/dist/_models/SystemState.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomationType = exports.RelationshipType = void 0;
var RelationshipType;
(function (RelationshipType) {
    RelationshipType["Partition"] = "devices/partition";
    RelationshipType["Lock"] = "devices/lock";
    RelationshipType["Camera"] = "video/camera";
    RelationshipType["GarageDoor"] = "devices/garage-door";
    RelationshipType["Scene"] = "automation/scene";
    RelationshipType["Sensor"] = "devices/sensor";
    RelationshipType["Light"] = "devices/light";
    RelationshipType["Thermostat"] = "devices/thermostat";
    RelationshipType["GeoDevice"] = "geolocation/geo-device";
    RelationshipType["GeoFence"] = "geolocation/fence";
    RelationshipType["SystemConfig"] = "systems/configuration";
    RelationshipType["System"] = "systems/system";
    RelationshipType["State"] = "devices/state-info";
})(RelationshipType || (exports.RelationshipType = RelationshipType = {}));
var AutomationType;
(function (AutomationType) {
    AutomationType["PeakProtect"] = "automation/peak-protect";
    AutomationType["RuleSuggestion"] = "automation/rules/ruleSuggestion";
})(AutomationType || (exports.AutomationType = AutomationType = {}));
